layers/parameter_validation_utils.cpp

```diff
@@ -4504,6 +4504,447 @@ bool StatelessValidation::manual_PreCallValidateUpdateDescriptorSets(VkDevice de
+                initial_layout == VK_IMAGE_LAYOUT_STENCIL_READ_ONLY_OPTIMAL) {
+                vuid = use_rp2 ? "VUID-VkAttachmentDescription2-format-03290" : "VUID-VkAttachmentDescription-format-03290";
+                skip |= LogError(device, vuid,
+                                 "%s: pCreateInfo->pAttachments[%" PRIu32
+                                 "].initialLayout must not be "
+                                 "VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL or"
+                                 "VK_IMAGE_LAYOUT_STENCIL_READ_ONLY_OPTIMAL",
+                                 func_name, i);
+            }
+            if (final_layout == VK_IMAGE_LAYOUT_STENCIL_ATTACHMENT_OPTIMAL ||
+                final_layout == VK_IMAGE_LAYOUT_STENCIL_READ_ONLY_OPTIMAL) {
+                vuid = use_rp2 ? "VUID-VkAttachmentDescription2-format-03291" : "VUID-VkAttachmentDescription-format-03291";
+                skip |= LogError(device, vuid,
+                                 "%s: pCreateInfo->pAttachments[%" PRIu32
+                                 "].finalLayout must not be "
+                                 "VK_IMAGE_LAYOUT_STENCIL_ATTACHMENT_OPTIMAL or "
+                                 "VK_IMAGE_LAYOUT_STENCIL_READ_ONLY_OPTIMAL",
+                                 func_name, i);
+            }
+        } else if (FormatIsStencilOnly(attachment_format) && !attachment_description_stencil_layout) {
+            if (initial_layout == VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL ||
+                initial_layout == VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_OPTIMAL) {
+                vuid = use_rp2 ? "VUID-VkAttachmentDescription2-format-06247" : "VUID-VkAttachmentDescription-format-03292";
+                skip |= LogError(device, vuid,
+                                 "%s: pCreateInfo->pAttachments[%" PRIu32
+                                 "].initialLayout must not be "
+                                 "VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL or"
+                                 "VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_OPTIMAL",
+                                 func_name, i);
+            }
+            if (final_layout == VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL ||
+                final_layout == VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_OPTIMAL) {
+                vuid = use_rp2 ? "VUID-VkAttachmentDescription2-format-06248" : "VUID-VkAttachmentDescription-format-03293";
+                skip |= LogError(device, vuid,
+                                 "%s: pCreateInfo->pAttachments[%" PRIu32
+                                 "].finalLayout must not be "
+                                 "VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL or "
+                                 "VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_OPTIMAL",
+                                 func_name, i);
+            }
+        }
+        if (attachment_description_stencil_layout) {
+            if (attachment_description_stencil_layout->stencilInitialLayout == VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL ||
```

Comment:
Create helper variables. Consistent with above code and helps readability.

```suggestion
            const VkImageLayout stencilInitialLayout = attachment_description_stencil_layout->stencilInitialLayout;
            const VkImageLayout stencilFinalLayout = attachment_description_stencil_layout->stencilFinalLayout;
            if (stencilInitialLayout == VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL ||
```

Reply:
This part was not touched in the copy over, but agree should be approved while here!